test/cases/compile_errors/reify_type_for_tagged_extern_union.zig

```diff
@@ -0,0 +1,34 @@
+const Tag = @Type(.{
+    .@"enum" = .{
+        .tag_type = u2,
+        .fields = &.{
+            .{ .name = "signed", .value = 0 },
+            .{ .name = "unsigned", .value = 1 },
+        },
+        .decls = &.{},
+        .is_exhaustive = true,
+    },
+});
+
+const Extern = @Type(.{
+    .@"union" = .{
+        .layout = .@"extern",
+        .tag_type = Tag,
+        .fields = &.{
+            .{ .name = "signed", .type = i32, .alignment = @alignOf(i32) },
+            .{ .name = "unsigned", .type = u32, .alignment = @alignOf(u32) },
+        },
+        .decls = &.{},
+    },
+});
+
+export fn entry() void {
+    const tagged: Extern = .{ .signed = -1 };
+    _ = tagged;
+}
+
+// error
+// backend=stage2
+// target=native
```

Comment:
Not a merge blocker, just a hint for the future: please don't write these two lines! They're defaults -- we should only explicitly specify these values when we need something special (e.g. a case is target-specific so `native` won't cut it). I'm trying to slowly remove these lines from the existing cases.